activemq-client/src/main/java/org/apache/activemq/ActiveMQSession.java

```diff
@@ -1956,12 +1978,17 @@ protected void send(ActiveMQMessageProducer producer, ActiveMQDestination destin
             //Set the "JMS" header fields on the original message, see 1.1 spec section 3.4.11
             message.setJMSDeliveryMode(deliveryMode);
             long expiration = 0L;
-            if (!producer.getDisableMessageTimestamp()) {
-                long timeStamp = System.currentTimeMillis();
+            long timeStamp = System.currentTimeMillis();
+            if (timeToLive > 0) {
+                expiration = timeToLive + timeStamp;
+            }
+
+            // TODO: AMQ-8500 - update this when openwire supports JMSDeliveryTime
```

Comment:
Might be good for this comment to reference the ActiveMQMessageTransformation related equivalent to avoid it being forgotten about.

Reply:
This is complete